### PR TITLE
README Overhaul

### PR DESCRIPTION
**Pull request by arpanlaha**
Updates `README.md` with the following:

- [x] Nonprofit background
- [x] Updated project description
- [x] Application architecture
- [x] Team showcase
- [x] License info

### CODE REVIEW

File: .github/workflows/api.yaml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: api
 
-on: push
+on:
```
Comment (arpanlaha) [inline .github/workflows/api.yaml]: I added this so that all my readme changes wouldn't unnecessarily set off the CI.

Comment (arpanlaha) [reply .github/workflows/api.yaml]: Also the CI icon is still yellow but just ignore it it isn't necessary for this PR.

**Comment by arpanlaha** [comment]
@yousefa00 is there anything you wanna add/change?

File: temp.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<h1>
+  FateMaker
+  <sub>
+    <a href="https://uiuc.hack4impact.org/"> Hack4Impac UIUC</a> ×
+    <a href="https://www.kidssaveocean.com/">Kids Save Ocean</a>
+  </sub>
+</h1>
```
Comment (chyku) [inline temp.html]: Is this supposed to be here?

Comment (arpanlaha) [reply temp.html]: ope

File: README.md [hunk 3]
```diff
@@ -1,7 +1,129 @@
...
+        <br />
+        <sub>Software Developer</sub>
+      </a>
+    </td>
+  </tr>
+</table>
+
+<h3 align="center">Architecture</h3>
+
+FateMaker uses React.js and Next.js for the frontend and Express.js and MongoDB for the backend, with Hack4Impact UIUC's [Product Infrastructure authentification server](https://github.com/hack4impact-uiuc/infra-authentication-server) used as well.
```
Comment (coconut750750) [inline README.md]: Isn't it spelled "authentication"?

**Pull request merged**